static/app/views/settings/project/server-side-sampling/utils/useSdkVersions.tsx

```diff
@@ -0,0 +1,66 @@
+import {useQuery} from '@tanstack/react-query';
+
+import {ResponseMeta} from 'sentry/api';
+import {t} from 'sentry/locale';
+import {Organization, Project} from 'sentry/types';
+import {SamplingDistribution, SamplingSdkVersion} from 'sentry/types/sampling';
+import handleXhrErrorResponse from 'sentry/utils/handleXhrErrorResponse';
+import useApi from 'sentry/utils/useApi';
+
+type Props = {
+  hasAccess: boolean;
+  organizationSlug: Organization['slug'];
+  projectId: Project['id'];
+  distribution?: SamplingDistribution;
+};
+
+export function useSdkVersions({
+  hasAccess,
+  distribution,
+  organizationSlug,
+  projectId,
+}: Props) {
+  const api = useApi();
+
+  const projectIds = [
+    projectId,
+    ...(distribution?.projectBreakdown?.map(
+      projectBreakdown => projectBreakdown.project_id
+    ) ?? []),
+  ];
+
+  const samplingSdkVersions = useQuery(
+    [
```

Comment:
We'll probably want some standardized way of creating query keys. I prefer an object at the second index since some keys might become optional and you get to name each key value.

The bigger issue is exporting a function or type that defines what the query key should look like. When you add mutations to the mix, you'll often want to create these keys in another location in order to update the cache (such as when you make a change and click save).

Reply:
> We'll probably want some standardized way of creating query keys. I prefer an object at the second index since some keys might become optional and you get to name each key value.

agree... I think it makes debugging easier

> The bigger issue is exporting a function or type that defines what the query key should look like. When you add mutations to the mix, you'll often want to create these keys in another location in order to update the cache (such as when you make a change and click save).

I haven't tried mutations yet but would love to see some examples, so I can better visualize the issue you are mentioning

Reply:
> I haven't tried mutations yet but would love to see some examples, so I can better visualize the issue you are mentioning

Yeah no problem, here's an example @priscilawebdev , hope it makes sense:

```ts
/* fetchProjectSettings.tsx */

const {data} = useQuery(['fetchProjectSettings', {org, projectSlug}], () => {
  // fetch project settings
});


/* saveProjectNameButton.tsx */

const queryClient = useQueryClient();
const {mutate} = useMutation(
  () => {
    // do the mutation
  },
  {
    // on success, need to update the cache so that the project name changes immediately.
    // if you don't do this, the project name will be out of date until it gets refetched
    onSuccess: data => {
      queryClient.setQueryData(['fetchProjectSettings', {org, projectSlug}], oldData => {
        return {
          ...oldData,
          name: data.name,
        };
      });
    },
  }
);
```

When something like this happens, you are using `['fetchProjectSettings', {org, projectSlug}]` in two different files. So if someone changes the query in `fetchProjectSettings.tsx` to include another property, we want to make sure that anything that updates that query also makes that change.

In the past, I've put my queries in a separate file that also exported a `makeFetchProjectSettingsQueryKey()` function.

Reply:
yes, it makes sense...thanks @malwilley